File: problems/CSES/Sorting_and_Searching/subarray_sums_i/subarray_sums_i.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n, target;
    cin >> n >> target;

    vector<int> arr(n);
    for (int i = 0; i < n; i++) cin >> arr[i];

    int sum = 0;
    int res = 0;
    
    int left = 0;

    for (int right = 0; right < n; right++) {
        sum += arr[right];

        while (sum > target) {
            sum -= arr[left];
            left++;
        }

        if (sum == target) {
            res++;
        }
    }

    cout << res << "\n";

    return 0;
}

File: problems/CSES/Sorting_and_Searching/subarray_sums_i/subarray_sums_i.py
import sys
# sys.setrecursionlimit(200000)

input = sys.stdin.readline

def main():
    n, target = map(int, input().split())
    arr = list(map(int, input().split()))

    sum, res = 0, 0

    left = 0

    for right in range(len(arr)):
        sum += arr[right]

        while sum > target:
            sum -= arr[left]
            left += 1

        if sum == target:
            res += 1

    print(res)

main()
